c10/core/TensorImpl.cpp

```diff
@@ -153,6 +163,8 @@ TensorImpl::TensorImpl(
 
   key_set = key_set | getAutocastRelatedKeySetFromBackend(k);
 
+  key_set = key_set.remove(DispatchKey::Python);
```

Comment:
might be worth a quick note on why we remove the python key specifically in TensorImpl constructors? I think I got it after looking at python_variable.cpp: Only tensors that are exposed to python should be capable of dispatching to python. E.g. if we call some C++ function that internally creates a view without exposing that view to python, we don't want it to do any python dispatching.

Reply:
```
// [Note: Python key removal]
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// In most constructors for TensorImpl, you will see Python key is removed from
// the passed in DispatchKeySet.  Why?
//
// INVARIANT: Python dispatch key is set iff PyObject for the Tensor has a
// nontrivial __torch_dispatch__ implementation.
//
// When a fresh TensorImpl is created, there is *no* PyObject (this only gets
// initialized lazily at the first point in time the Tensor passes into Python).
// So we would violate the invariant.
//
// In practice, what will happen shortly afterwards is that the TensorImpl
// will get its PyObject initialized by Tensor._make_subclass; at this point
// the Python dispatch key will be set and all is well.  The point is to delay
// the dispatch key setting until that point.
```